src/lbaf/Execution/lbsAlgorithmBase.py

```diff
@@ -29,35 +27,24 @@ def __init__(self, work_model: WorkModelBase, parameters: dict, logger: Logger,
         """
         # Assert that a logger instance was passed
         if not isinstance(logger, Logger):
-            get_logger().error(
-                f"Incorrect type {type(logger)} passed instead of Logger instance")
-            sys.excepthook = exc_handler
-            raise SystemExit(1)
+            raise TerseError(f"Incorrect type {type(logger)} passed instead of Logger instance")
```

Comment:
What is the purpose of this `TerseError()`? Reduce code replication?

Reply:
Hello @ppebay this is due to refactoring.
In the code this was not consistent. Sometimes we raised errors and each time we do that after setting the exception handler. For a developer this is not intuitive. We should set the handler only once for an application and raise errors and then use the handler to process it if needed. Thats why many logger errors followed by system exit commands have been replaced by errors raising. Because it seems more logical for an error to be fired as a Python exception si it can if needed be intercepted. I think it is a better practice.

TerseError means no stack trace Just a simple error message to display. This is Just a way to know in the handler what we want to display. An alternative could be to detect only error raised by LBAF to be displayed with only the message then we could remove the TerseError class.

Reply:
I see your point about the codebase not being consistent with exception throwing and error-ing out.

That being said, the intent at least was the following:
* any condition resulting in an ERROR (and logged as such) should cause the code to terminate immediately; the rationale being that if a type-I error (i.e. a "known error" that can tested as such) occurs, this means that any subsequent result produced by the code could be incorrect; as a result the code should not be allowed to pursue further;
* in contrast, a condition merely resulting in a WARNING (and logged as such) could be allowed to proceed after having communicated the message as you described in the second paragraph above.

So, my point is that errors should always result in code termination. What you call a `TerseError` more resembles a warning in my definition.

Reply:
TerseError removed and consistency added :
- logger error followed by SystemExit  everywhere we want to terminate with an ERROR.
- exception handler removed